### PR TITLE
Make map markers go to drop when clicked

### LINKED ISSUE #88: drop-markers in index view should link to individual drops
Ideally, the markers should be 'clickable': 
they should direct to the drop show view 
A problem could be that if there’s only one marker shown when there are actually several drops because they are in the same place, like e.g. at Soundcloud. (There are at least 4 drops located at SC, but there's only one marker). Maybe that one marker containing several drops could direct to a list of drops? 
Maybe it is more complicated as there is probably not only one marker but several on top of each other, like a pile. But you can only see the one on top.

### PR DESCRIPTION
**Pull request by berlintam**
closes https://github.com/sounddrop/sounddrop/issues/88

(I did not address this issue of what to do when there are multiple drops in one location. I think this is probably part of a larger discussion & can be addressed in detail in another PR). 

I'm not a big javascripter, so maybe someone else can take a look at this? I'm pretty sure L27 of map.js is not super elegant... @axlwaii feel free to comment on this if you want!

### CODE REVIEW

**Comment by axlwaii** [comment]
Looks good so far 👍 

The only the thing that bugs me is that `coordinatesList` is defined outside the maps module.
We could try to extract the coordinates directly from the html via `data-attribute`.

``` html
<!-- drops#index -->
<div id="map" class="map-index" data-drops=<%= @drops.map{ |drop| {lat: drop.latitude.to_f, lng: drop.longitude.to_f, drop: drop.id} }.to_json.html_safe %>></div>

<!-- drops#show -->
<div id='map' class='map' data-drops=<%= [ { lat: @drop.latitude.to_f, lng: @drop.longitude.to_f, id: @drop.id}].to_json.html_safe %>>
</div>
```

``` javascript
// map.js
var Map = {
  map: null,
  setMarker: function() {},
  init: function() {
    var dropLat, 
        dropLong,
        coordinateList,
        map;

    map = document.getElementById('map');
    coordinateList = map.dataset.drops ? JSON.parse(map.dataset.drops) : [];

// ....
```

**Comment by berlintam** [comment]
cool, thanks @axlwaii! I'll give it a shot.

**Comment by aydamacink** [comment]
Alt Text

**Comment by aydamacink** [comment]
alt text

**Comment by aydamacink** [comment]
alt text

File: app/assets/javascripts/map.js [hunk 1]
```diff
@@ -22,26 +22,19 @@ var Map = {
...
     for (var i = 0; i < coordinateList.length; i++ ){
-       var marker = new google.maps.Marker({
-        position: coordinateList[i],
-         icon: icon,
-        map: Map.map
+      var marker = new google.maps.Marker({
+        position: {'lat': coordinateList[i]['lat'], 'lng': coordinateList[i]['lng']},
+        icon: icon,
+        map: Map.map,
+        dropUrl: location.protocol + '//' + location.host + '/drops/' + coordinateList[i]['drop']
```
Comment (aydamacink) [inline app/assets/javascripts/map.js:32]: here the third key is called `drop`

File: app/views/drops/show.html.erb [hunk 2]
```diff
@@ -10,7 +10,7 @@
   <div class="artwork">
     <img class="container-image" src="<%=@drop.image_from_track%>">
     <%= render partial: "load_playbuttons" %>
-    <div id='map' class='map'></div>
+    <div id='map' class='map' data-drops=<%= [ { lat: @drop.latitude.to_f, lng: @drop.longitude.to_f, id: @drop.id}].to_json.html_safe %>></div>
```
Comment (aydamacink) [inline app/views/drops/show.html.erb:13]: here the third key is called `id`

File: app/views/welcome/index.html.erb [hunk 3]
```diff
@@ -7,11 +7,7 @@
     <p>With SoundDrop you can leave an audible mark anywhere and share it with friends. Wish to leave a playlist along your favourite bike route through town? Or create a guided tour for a museum exhibition? Want to tell a personal story about a famous landmark, or guide your friends to your favourite spots in a foreign city? SoundDrop let's you share the sound.</p>
 
     <div class='container-map-index'>
-      <div id="map" class="map-index"></div>
-
-      <script type="text/javascript">
-      var coordinateList = JSON.parse('<%= @drops.map{ |drop| { lat: drop.latitude.to_f, lng: drop.longitude.to_f} }.to_json.html_safe %>');
-      </script>
+      <div id="map" class="map-index" data-drops=<%= @drops.map{ |drop| { lat: drop.latitude.to_f, lng: drop.longitude.to_f, drop: drop.id } }.to_json.html_safe %>></div>
```
Comment (aydamacink) [inline app/views/welcome/index.html.erb:10]: here the third key is called `drop`

**Pull request merged**